### PR TITLE
Fix attention vizualizer

### PR DESCRIPTION
**Pull request by molbap**
# What does this PR do?

I noticed a few models were broken with the visualizer, because it relied on `_update_causal_mask`, and we changed the interface quite a bit. I tested it, added a test for PaliGemma and Llama3.2 as well. 

Also added a test to make the tool a tad more durable!

### CODE REVIEW

File: src/transformers/utils/attention_visualizer.py [hunk 1]
```diff
@@ -207,13 +208,23 @@ def visualize_attention_mask(self, input_sentence: str, suffix=""):
...
+            input_embeds=input_embeds,
             attention_mask=attention_mask,
-            input_tensor=attention_mask.to(self.model.dtype),
-            cache_position=torch.arange(attention_mask.shape[1]),
+            cache_position=cache_position,
             past_key_values=None,
-            **kwargs,
-        ).bool()
+        )
+
```
Comment (ArthurZucker) [inline src/transformers/utils/attention_visualizer.py:223]: nice! We van merge as is, the best way to get the actual mask created by the model is to monkey patch decoder layer's forward...... a little bit annouying

**Review by ArthurZucker — APPROVED** [review]
thanks for adding a test as well!

File: src/transformers/utils/attention_visualizer.py (hunk 1 above)
Comment (molbap) [reply src/transformers/utils/attention_visualizer.py:223]: arg, yes. I'll push the processor files for the tests to hf internal testing and merge it then, thanks!
